File: e2e/elixir/test/cookies_test.exs
# E2e tests for category: cookies
defmodule E2e.CookiesTest do
  use ExUnit.Case, async: true

  describe "cookies_per_domain" do
    test "Isolates cookies per domain during crawl" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(length(result.cookies)) == 1
      assert String.contains?(result.cookies, "domain_cookie")
    end
  end

  describe "cookies_persistence" do
    test "Maintains cookies across multiple crawl requests" do
      result = Kreuzcrawl.scrape!()
      assert String.contains?(result.cookies, "session")
    end
  end

  describe "cookies_set_cookie_response" do
    test "Respects Set-Cookie header from server responses" do
      result = Kreuzcrawl.scrape!()
      assert String.contains?(result.cookies, "tracking")
    end
  end
end

File: e2e/rust/tests/cookies_test.rs
//! E2e tests for category: cookies

use kreuzcrawl::scrape;
use kreuzcrawl::create_engine;

#[tokio::test]
async fn test_cookies_per_domain() {
    // Isolates cookies per domain during crawl
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'cookies.length' not available on result type
    // skipped: field 'cookies' not available on result type
}

#[tokio::test]
async fn test_cookies_persistence() {
    // Maintains cookies across multiple crawl requests
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'cookies' not available on result type
}

#[tokio::test]
async fn test_cookies_set_cookie_response() {
    // Respects Set-Cookie header from server responses
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'cookies' not available on result type
}
